ts/features/idpay/payment/screens/IDPayPaymentAuthorizationScreen.tsx

```diff
@@ -59,39 +69,207 @@ const IDPayPaymentAuthorizationScreen = () => {
+  const renderContent = () => {
+    if (isLoading) {
+      return <TransactionDataSkeleton />;
+    }
+    if (transactionData !== undefined) {
+      return <ExistingTransactionDataContent data={transactionData} />;
+    }
+    // TODO:: error page, will be added in IOBP-6
+    return <></>;
+  };
 
   return (
     <BaseScreenComponent
       headerTitle="Autorizza operazione"
       contextualHelp={emptyContextualHelp}
+      goBack={false}
     >
-      <LoadingSpinnerOverlay isLoading={isLoading} loadingOpacity={100}>
-        <SafeAreaView style={IOStyles.flex}>
-          <View style={IOStyles.flex}>
-            <ContentWrapper>{content}</ContentWrapper>
-          </View>
-          <FooterWithButtons
-            type="TwoButtonsInlineHalf"
-            leftButton={{
-              title: "Annulla",
-              bordered: true,
-              onPress: handleCancel,
-              disabled: isUpserting
-            }}
-            rightButton={{
-              title: isUpserting ? "" : "Conferma",
-              onPress: handleConfirm,
-              isLoading: isUpserting,
-              disabled: isUpserting
-            }}
-          />
-        </SafeAreaView>
-      </LoadingSpinnerOverlay>
+      <SafeAreaView style={IOStyles.flex}>
+        <View style={IOStyles.flex}>
+          <ContentWrapper>
+            <H1>{I18n.t("idpay.payment.authorization.header")}</H1>
+            <VSpacer size={24} />
+            {renderContent()}
+          </ContentWrapper>
+        </View>
+        <FooterWithButtons
+          type="TwoButtonsInlineHalf"
+          leftButton={{
+            title: I18n.t("global.buttons.cancel"),
+            bordered: true,
+            onPress: handleCancel,
+            disabled: isUpserting
+          }}
+          rightButton={{
+            title: isUpserting ? "" : I18n.t("global.buttons.confirm"),
+            onPress: handleConfirm,
+            isLoading: isUpserting || isLoading,
+            disabled: isUpserting || isLoading
+          }}
+        />
+      </SafeAreaView>
     </BaseScreenComponent>
   );
 };
 
-export type { IDPayPaymentAuthorizationScreenRouteParams };
+const ExistingTransactionDataContent = ({
+  data
+}: {
+  data: NonNullable<AuthPaymentResponseDTO>;
+}) => {
+  const getValueOrDash = (value: string | undefined) =>
+    pipe(
+      value,
+      O.fromNullable,
+      O.getOrElse(() => "-")
+    );
+  const amountCents = formatNumberCurrency(data.amountCents / 100);
+  const reward = pipe(
+    data.reward,
+    O.fromNullable,
+    O.fold(
+      () => "-",
+      data => formatNumberCurrency(data / 100)
+    )
+  );
+  const businessName = getValueOrDash(data.businessName);
+  const initiativeName = getValueOrDash(data.initiativeName);
+  const date = pipe(
+    data.trxDate,
+    O.fromNullable,
+    O.fold(
+      () => "-",
+      date => format(new Date(date), "D MMMM YYYY, HH:mm")
+    )
+  );
```

Comment:
Maybe we can create an utility function to format dates or get a default if not available, [like the `formatNumberCurrencyOrDefault` function](https://github.com/pagopa/io-app/blob/943665e9d01b7189e584acf8ac3f080e39898854/ts/features/idpay/common/utils/strings.ts#LL8C47-L8C47).

Something like this:

```js
export const formatDateOrDefault = (
  input: date | undefined,
  format: string,
  defaultValue: string = "-"
) =>
  pipe(
    input,
    O.fromNullable,
    O.map(date => format(date, "D MMMM YYYY, HH:mm")),
    O.getOrElse(() => defaultValue)
  );
```

What do you think?

Reply:
I think it wouldn't be so useful as of right now, since to me it seems like a pretty separated use case from the one you mentioned; having said this, do you still believe it could be that useful?

Reply:
I think that would be pretty useful. We can replace 10-15 lines of codes with one function everywhere. But it's not mandatory, this is just an improvement.